### PR TITLE
67 migrate to typescript

### PR DESCRIPTION
**Pull request by denolia**
In this request the first step towards Typescript is made:
* Typescript is added to deps
* Babel is replaced with ts-loader
* `Any` and `@ts-expect-error` are added to comply with Typescript requirements - this is to be improved and fixed later in separate requests

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -2,11 +2,13 @@ import ReactSelectorQuery from './src/resq'
 import { waitToLoadReact } from './src/waitToLoadReact'
 import { findReactInstance } from './src/utils'
 
-function doQuery(selector, method, element) {
+function doQuery(selector: any, method: any, element: any) {
+    // @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'global'.
```
Comment (baruchvlz) [inline index.ts:6]: `global` is also a reference to `window`, so we can use that instead.

File: .eslintrc.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+module.exports = {
```
Comment (baruchvlz) [inline .eslintrc.ts:1]: Can't we use TypeScript exports instead?

File: package.json [hunk 3]
```diff
@@ -28,16 +28,25 @@
         "@babel/core": "7.4.0",
         "@babel/plugin-transform-runtime": "7.4.0",
         "@babel/preset-env": "7.4.1",
+        "@babel/preset-typescript": "^7.12.1",
```
Comment (baruchvlz) [inline package.json:31]: I would prefer to move away from babel if we can build with `tsc`.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+     "jsx": "react",                          /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
```
Comment (baruchvlz) [inline tsconfig.json:12]:
```suggestion
    // "jsx": "react",                          /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
```

**Review by baruchvlz — COMMENTED** [review]
I know this is still a WIP, just some comments.

**Comment by baruchvlz** [comment]
Changed the target branch to v2 in case we have another minor or patch release.

File: index.ts (hunk 1 above)
Comment (denolia) [reply index.ts:6]: Thanks!
Changed that to `window` and added type declarations for react stuff

File: package.json (hunk 3 above)
Comment (denolia) [reply package.json:31]: ok, let me try to remove it

File: .eslintrc.ts (hunk 2 above)
Comment (denolia) [reply .eslintrc.ts]: looks like it is more natural to use json

**Comment by gurkerl83** [comment]
Is it somehow possible to use actual react types instead of "any" everywhere? Thx!

**Comment by denolia** [comment]
> Is it somehow possible to use actual react types instead of "any" everywhere? Thx!

That is the final goal. 
This request is only the first step towards it - to introduce Typescript and make build and test working with it.

**Review by baruchvlz — APPROVED** [review]

**Pull request merged**